tests/unittests/test_sync.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+from unittest import mock
+from parameterized import parameterized
+from tap_quickbooks.client import QuickbooksClient
+from tap_quickbooks.sync import do_sync
+
+class Schema:
+    """Mocked Schema"""
+
+    def __init__(self,stream_name) -> None:
+        self.stream = stream_name
+
+    def to_dict(self):
+        return {'stream':self.stream}
+
+class Stream:
+    """Mocked Stream """
+
+    def __init__(self,stream_name) -> None:
+        self.tap_stream_id = stream_name
+        self.schema = Schema(stream_name)
+        self.metadata = {}
+
+class MockCatalog:
+    """Mocked Catalog """
+
+    def __init__(self,stream) -> None:
+        self.stream = stream
+
+    def get_selected_streams(self,state):
+        for stream in self.stream:
+            yield Stream(stream)
+
```

Comment:
One line is left to cover: 
```
raise Exception("Attempted to sync unknown stream {}".format(stream_id))
```
when stream_object is None

Reply:
Added unittest to cover the particular line.